direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java

```diff
@@ -84,39 +84,57 @@
   @Nullable private transient Cluster cluster;
   /** Session we are connected to. */
   @Nullable private transient Session session;
+  /** Quorum for both reads and writes. */
+  private ConsistencyLevel consistencyLevel;
 
   @SuppressWarnings({"unchecked", "rawtypes"})
   public CassandraDBAccessor(EntityDescriptor entityDesc, URI uri, Map<String, Object> cfg) {
 
     super(entityDesc, uri);
 
-    Object factoryName = cfg.get(CQL_FACTORY_CFG);
-    this.cqlFactoryName =
-        factoryName == null ? DefaultCqlFactory.class.getName() : factoryName.toString();
+    this.cqlFactoryName = getCqlFactoryName(cfg);
+    this.batchParallelism = getBatchParallelism(cfg);
+    this.converter = getStringConverter(cfg);
+    this.consistencyLevel = getConsistencyLevel(cfg);
+    initializeCqlFactory();
+  }
 
-    Object tmp = cfg.get(CQL_PARALLEL_SCANS);
+  private String getCqlFactoryName(Map<String, Object> cfg) {
+    Object tmp = cfg.get(CQL_FACTORY_CFG);
+    return tmp == null ? DefaultCqlFactory.class.getName() : tmp.toString();
+  }
+
+  private int getBatchParallelism(Map<String, Object> cfg) {
+    final Object tmp = cfg.get(CQL_PARALLEL_SCANS);
+    final int ret;
     if (tmp != null) {
-      batchParallelism = Integer.parseInt(tmp.toString());
+      ret = Integer.parseInt(tmp.toString());
     } else {
-      batchParallelism = Runtime.getRuntime().availableProcessors();
-    }
-
-    if (batchParallelism < 2) {
-      throw new IllegalArgumentException(
-          "Batch parallelism must be at least 2, got " + batchParallelism);
+      ret = Math.max(2, Runtime.getRuntime().availableProcessors());
     }
+    Preconditions.checkArgument(ret >= 2, "Batch parallelism must be at least 2, got %s", ret);
+    return ret;
+  }
 
+  private StringConverter<?> getStringConverter(Map<String, Object> cfg) {
+    Object tmp;
     tmp = cfg.get(CQL_STRING_CONVERTER);
-    StringConverter<String> c = StringConverter.getDefault();
+    StringConverter<?> c = StringConverter.getDefault();
     if (tmp != null) {
       try {
         c = Classpath.newInstance(tmp.toString(), StringConverter.class);
       } catch (Exception ex) {
         log.warn("Failed to instantiate type converter {}", tmp, ex);
       }
     }
-    this.converter = (StringConverter) c;
-    initializeCqlFactory();
+    return c;
+  }
+
+  private ConsistencyLevel getConsistencyLevel(Map<String, Object> cfg) {
+    return Optional.ofNullable(cfg.get("consistency-leve"))
+        .map(Object::toString)
+        .map(ConsistencyLevel::valueOf)
+        .orElse(ConsistencyLevel.QUORUM);
```

Comment:
Nechces spise `LOCAL_QUORUM `

Reply:
No, `LOCAL_QUORUM` does not work with replication strategy `SimpleStrategy`. It is meant to be for multi-cluster setups, which should not be the default.